tensorboard/webapp/widgets/custom_modal/custom_modal_test.ts

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {CustomModalComponent} from './custom_modal_component';
+import {CommonModule} from '@angular/common';
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Output,
+  ViewChild,
+} from '@angular/core';
+
+function waitFrame() {
+  return new Promise((resolve) => window.requestAnimationFrame(resolve));
+}
+
+@Component({
+  selector: 'testable-modal',
+  template: `<custom-modal #modal (onOpen)="setOpen()" (onClose)="setClosed()">
+    <div>My great content</div>
+  </custom-modal> `,
+})
+class TestableComponent {
+  @ViewChild('modal', {static: false})
+  modalComponent!: CustomModalComponent;
+
+  @ViewChild('content', {static: false})
+  content!: ElementRef;
+
+  isOpen = false;
+
+  @Output() onOpen = new EventEmitter();
+  @Output() onClose = new EventEmitter();
+
+  setOpen() {
+    this.isOpen = true;
+    this.onOpen.emit();
+  }
+
+  setClosed() {
+    this.isOpen = false;
+    this.onClose.emit();
+  }
+
+  close() {
+    this.modalComponent.close();
+  }
+
+  getContentStyle() {
+    return (this.modalComponent as any).content.nativeElement.style;
+  }
+
+  public openAtPosition(position: {x: number; y: number}) {
+    this.modalComponent.openAtPosition(position);
+  }
+}
+
+describe('custom modal', () => {
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [TestableComponent, CustomModalComponent],
+      imports: [CommonModule],
+    }).compileComponents();
+  });
+
+  it('waits a frame before emitting onOpen or onClose', async () => {
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.detectChanges();
+    fixture.componentInstance.openAtPosition({x: 0, y: 0});
+    expect(fixture.componentInstance.isOpen).toBeFalse();
```

Comment:
Can we move this expect to before you call openAtPosition? We are trying to test that it starts closed not that we have to wait for the requestAnimationFrame to trigger.

Reply:
I have another test which checks that it starts closed. This test IS specifically ensuring that it waits for `requestAnimationFrame`